llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6082,15 +6083,20 @@ SwitchLookupTable::SwitchLookupTable(
           LinearMappingPossible = false;
           break;
         }
-        Wrapped |=
-            Dist.isStrictlyPositive() ? Val.sle(PrevVal) : Val.sgt(PrevVal);
+        NMon |= Dist.isStrictlyPositive() ? Val.sle(PrevVal) : Val.sgt(PrevVal);
       }
       PrevVal = Val;
     }
     if (LinearMappingPossible) {
       LinearOffset = cast<ConstantInt>(TableContents[0]);
       LinearMultiplier = ConstantInt::get(M.getContext(), DistToPrev);
-      LinearMapValWrapped = Wrapped;
+      bool MayMWrap = false, MayOWrap = false;
+      APInt M = LinearMultiplier->getValue(), O = LinearOffset->getValue();
+      APInt Res = O.sadd_ov(
```

Comment:
Is the sadd_ov check here needed as well? If yes, is it possible to add a test case that needs it?